jquery.flot.js

```diff
@@ -1949,7 +1962,68 @@ Licensed under the MIT license.
+                        var intersecting = false;
+                        if (previousLabelBox) {
+                            intersecting |= areRectanglesIntersecting(
+                                  previousLabelBox.x, previousLabelBox.y, previousLabelBox.x + previousLabelBox.width, previousLabelBox.y + previousLabelBox.height,
+                                  newLabelBox.x, newLabelBox.y, newLabelBox.x + newLabelBox.width, newLabelBox.y + newLabelBox.height);
+                        }
+                        if (lastLabelBox) {
+                            intersecting |= areRectanglesIntersecting(
+                                  lastLabelBox.x, lastLabelBox.y, lastLabelBox.x + lastLabelBox.width, lastLabelBox.y + lastLabelBox.height,
+                                  newLabelBox.x, newLabelBox.y, newLabelBox.x + newLabelBox.width, newLabelBox.y + newLabelBox.height);
+                        }
+                        return intersecting;
+                    },
+                    drawAxisLabel = function (index) {
+                        tick = axis.ticks[index];
+                        if (!tick.label || tick.v < axis.min || tick.v > axis.max ||
+                            (axis.options.showTickLabels == 'none') ||
+                            (axis.options.showTickLabels == 'endpoints' && !(index == 0 || index == axis.ticks.length - 1)) ||
+                            (axis.options.showTickLabels == 'major' && (index == 0 || index == axis.ticks.length - 1))) {
+                            return;
+                        }
+
+                        info = surface.getTextInfo(layer, tick.label, font, null, maxWidth);
+
+                        if (axis.direction == "x") {
+                            halign = "center";
+                            x = plotOffset.left + axis.p2c(tick.v);
+                            if (axis.position == "bottom") {
+                                y = box.top + box.padding;
+                            } else {
+                                y = box.top + box.height - box.padding;
+                                valign = "bottom";
+                            }
+                            newLabelBox = {x: x - info.width / 2, y: y, width: info.width, height: info.height}
+                        } else {
+                            valign = "middle";
+                            y = plotOffset.top + axis.p2c(tick.v);
+                            if (axis.position == "left") {
+                                x = box.left + box.width - box.padding;
+                                halign = "right";
+                            } else {
+                                x = box.left + box.padding;
+                            }
+                            newLabelBox = {x: x, y: y - info.height / 2, width: info.width, height: info.height}
+                        }
+
+                        if (isIntersectingWithOtherLabelsBoxes(newLabelBox)) {
```

Comment:
hmmm .. I'm not too sure. this function is too complicated. we need to rework it.